### PR TITLE
Adding the samples for handling EntityExpansion

### PR DESCRIPTION
**Pull request by Prakhash**
1. With xerces without the SecurityManager
2. With xerces with the SecurityManager
3. WSO2 recommended implementation

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: samples/TestEntityExpansionLimit/TestEntityExpansionLimit.iml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ayomawdb) [inline samples/TestEntityExpansionLimit/TestEntityExpansionLimit.iml]: We should not commit IMLs into Github

File: samples/TestEntityExpansionLimit/src/main/java/xxe/DemoWithSecurityManager.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package xxe;
```
Comment (ayomawdb) [inline samples/TestEntityExpansionLimit/src/main/java/xxe/DemoWithSecurityManager.java]: Can we use the standard package naming for this. This should be 

org.wso2.security.tools.samples.xee.jaxb
org.wso2.security.tools.samples.xee.dom

File: samples/TestEntityExpansionLimit/pom.xml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            <version>1.1.1</version>
+        </dependency>
+        <dependency>
+            <groupId>xerces</groupId>
+            <artifactId>xercesImpl</artifactId>
+            <version>2.9.1</version>
+        </dependency>
+    </dependencies>
+
+
```
Comment (ayomawdb) [inline samples/TestEntityExpansionLimit/pom.xml]: Please include checkstyle plugin to all the WSO2 maintained maven poms: 

Example: https://github.com/ballerina-lang/ballerina/blob/7a2723b6d0b5e0695e55555c0f16fefe099af963/pom.xml#L902

File: samples/TestEntityExpansionLimit/TestEntityExpansionLimit.iml (hunk 1 above)
Comment (Prakhash) [reply samples/TestEntityExpansionLimit/TestEntityExpansionLimit.iml]: Deleted the file

File: samples/TestEntityExpansionLimit/src/main/java/xxe/DemoWithSecurityManager.java (hunk 2 above)
Comment (Prakhash) [reply samples/TestEntityExpansionLimit/src/main/java/xxe/DemoWithSecurityManager.java]: Done

File: samples/TestEntityExpansionLimit/pom.xml (hunk 3 above)
Comment (Prakhash) [reply samples/TestEntityExpansionLimit/pom.xml]: Done

File: samples/TestEntityExpansionLimit/src/main/java/xxe/DemoWithSecurityManager.java (hunk 2 above)
Comment (ayomawdb) [reply samples/TestEntityExpansionLimit/src/main/java/xxe/DemoWithSecurityManager.java]: Package name is still "xxe.dom" 

Also, this is entity expansion not "external entity injection (XXE)". Hence, let's invent a name (xee) or use "entityexpansion" in package.

**Comment by ayomawdb** [comment]
Please review the comment on the folder structure. 

> Project can be stored in
"security-tools/samples/xml/xee/jaxb"
and
"security-tools/samples/xml/xee/dom"

**Comment by thariyarox** [comment]
Need to move the samples folder into https://github.com/wso2/security-tools/tree/master/internal folder. Please re-send the PR with above modification.

**Pull request closed without merging**